### PR TITLE
Migrate examples to junit5

### PR DESCRIPTION
**Pull request by samed-bicer**
migrate examples from junit4 to junit5
#6921

### CODE REVIEW

**Comment by samed-bicer** [comment]
hi @eddumelendez, examples are migrated to junit5. Could you please review the PR, thanks!

**Review by eddumelendez — COMMENTED** [review]
Using JUnit 5, public can be removed from test classes and methods. Please apply this change.

Thanks!

**Comment by samed-bicer** [comment]
> Using JUnit 5, public can be removed from test classes and methods. Please apply this change.
> 
> Thanks!

It has been removed, thanks. 
The [ci job for tubo-mode](https://github.com/testcontainers/testcontainers-java/actions/runs/5968497809/job/16202730271?pr=7417) is failing with timeout, is there any solution for that?

File: examples/spring-boot-kotlin-redis/src/test/kotlin/com/example/redis/AbstractIntegrationTest.kt [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 import org.springframework.boot.test.util.TestPropertyValues
 import org.springframework.context.ApplicationContextInitializer
 import org.springframework.context.ConfigurableApplicationContext
 import org.springframework.test.context.ContextConfiguration
-import org.springframework.test.context.junit4.SpringRunner
+import org.springframework.test.context.junit.jupiter.SpringExtension
 import org.testcontainers.containers.GenericContainer
 
-@RunWith(SpringRunner::class)
+@ExtendWith(SpringExtension::class)
```
Comment (eddumelendez) [inline examples/spring-boot-kotlin-redis/src/test/kotlin/com/example/redis/AbstractIntegrationTest.kt:13]: it is not needed. `@SpringBootTest` already use it internally.

```suggestion
@ExtendWith(SpringExtension::class)
```

File: examples/spring-boot/src/test/java/com/example/AbstractIntegrationTest.java [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.DynamicPropertyRegistry;
 import org.springframework.test.context.DynamicPropertySource;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
 import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.utility.DockerImageName;
 
-@RunWith(SpringJUnit4ClassRunner.class)
+@ExtendWith(SpringExtension.class)
```
Comment (eddumelendez) [inline examples/spring-boot/src/test/java/com/example/AbstractIntegrationTest.java:13]:
```suggestion
```

File: examples/spring-boot-kotlin-redis/src/test/kotlin/com/example/redis/RedisTest.kt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package com.example.redis
 
 import org.hamcrest.CoreMatchers.containsString
-import org.junit.Test
+import org.junit.jupiter.api.Test;
```
Comment (eddumelendez) [inline examples/spring-boot-kotlin-redis/src/test/kotlin/com/example/redis/RedisTest.kt:4]:
```suggestion
import org.junit.jupiter.api.Test
```

File: examples/spring-boot-kotlin-redis/src/test/kotlin/com/example/redis/RedisApplicationTests.kt [hunk 4]
```diff
@@ -1,6 +1,6 @@
 package com.example.redis
 
-import org.junit.Test
+import org.junit.jupiter.api.Test;
```
Comment (eddumelendez) [inline examples/spring-boot-kotlin-redis/src/test/kotlin/com/example/redis/RedisApplicationTests.kt:3]:
```suggestion
import org.junit.jupiter.api.Test
```

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
Thanks for your contribution @samed-bicer !

**Pull request merged**